lib/lightning/kafka_triggers/message_handling.ex

```diff
@@ -72,4 +74,14 @@ defmodule Lightning.KafkaTriggers.MessageHandling do
         {:error, message}
     end
   end
+
+  defp convert_for_serialisation(%{headers: headers} = metadata) do
+    metadata |> Map.merge(%{headers: list_of_lists(headers)})
+  end
+
+  defp list_of_lists(list) do
+    list
+    |> Enum.reduce([], fn tuple, acc -> [tuple |> Tuple.to_list() | acc] end)
```

Comment:
I am not familiar with the input but id it needs to transform every tuple on a list into a list it can be done with a Enum.map. it would avoid the reverse.

Reply:
@rorymckinley I first changed it to:

```elixir
metadata |> Map.merge(%{headers: Enum.into(headers, [], &Tuple.to_list/1)})
```

Then I thought, thats a bit obnoxious; I've changed it to this:

```elixir
Map.put(metadata, :headers, Enum.map(headers, &Tuple.to_list/1))
```

👌